### ISSUE TITLE
From reddint user /u/sansanity: Failed to read QR code from bitcoin.com and Bitpay on Send screen

### ISSUE THREAD

**Issue by cculianu** [cculianu/Electron-Cash#1]
Issue as reported:

The UI looks really clean, I like it a lot so far.

Ran into a problem with QR scanning on the Send function. I tested it against the Bitcoin.com and BitPay wallets using the following steps:

1) Brought up Receive QR Code on Bitcoin.com Wallet

2) Selected Send in Electron Cash

3) Clicked on QR Scanner

4) Highlighted QR code on Bitcoin.com wallet

At that point Electron seemed to recognize the code and went back to the Send page, but the address wasn't populated. I tried it with legacy and cash addrs on both of the wallets.

Let me know if you need more info, or want me to try anything.

**Comment by hcwagner** [comment]
Just a little extra info, non of the fields populated if I used a Receive QR code that specified an amount or fee.

**Comment by ldwrs** [comment]
I had the same problem scanning a cashaddr QR code, but when I scanned a legacy QR code (from explorer.bitcoin.com) it did work and the address was populated.

**Comment by cculianu** [comment]
Likely there are multiple "URL formats" (yes, these things get encoded as URLs) that I was unaware of and need to implement.

Thanks so much to the both of you for reporting this and giving me more info.  This is great.  Like I said previously -- this is exactly the sort of thing I was hoping for when asking others to look and test.

I'll look into it either today or tomorrow and expect a fix soon.  It's likely something simple.

Thanks!

EDIT: OOPS, Closed the issue by mistake.  Heh.. reopened until fixed.  Sorry about that!

**Comment by cculianu** [comment]
Yeah thanks guys.  I fixed the bug.  Yep... I had broken the QR scanning code late 2 nights ago.  Internally it was throwing an error.  I am a little embarrassed I broke it and didn't notice.  

Thanks so much for testing and for catching this! This was a relatively serious (or annoying) bug and I'm glad it's caught!!

It's addressed in the latest build which has gone up on TestFlight just now.  Open the TestFlight app to get the latest build.

It should work now on any style payment requests (Including bitpay!).

I tested it and it works.

If you guys write back and tell me you don't have problems, I can go ahead and close the issue. Will wait on you guys though.  Thanks so much!
